### ISSUE TITLE
Add a git-ratchet formatter

### ISSUE THREAD

**Issue by jvdl** [eslint/eslint#11853]
**The version of ESLint you are using.**
Any

**The problem you want to solve.**
I want to use ESLint to report on problems in my existing codebase and am fine with existing violations existing, however I would like to prevent additional violations from being added to the codebase.

For example, I would like to enable the "semi" rule so that semicolons are required, and currently have 100 violations in the codebase. However if someone adds a line of code without a semicolon so that there are now 101 violations, I would like my CI build to fail.

**Your take on the correct solution to problem.**
[git-ratchet](http://gitratchet.org/) can be used to keep track of measures in a codebase, it can then report that a metric has increased and a CI build can be failed.

Its reporting format is very simple: `rule-name,N` where `N` is the number of violations for that metric. 

e.g.:

```
semi,100
curly,8
comma-dangle,19
```
Note that this reports on total violations found, not on violations per file.

**Are you willing to submit a pull request to implement this change?**
Yes, I'd just like to gauge the interest and suitability.

**Comment by g-plane** [comment]
Hey @jvdl , thanks for the issue.

We recommend you to write a custom formatter: https://eslint.org/docs/developer-guide/working-with-custom-formatters

**Comment by jvdl** [comment]
Hi @g-plane, I have a custom formatter for it in the project I use it in already, I wanted to gauge the interest in having it included as a formatter in the ESLint project. If the interest is there, I'm happy to open a PR for it :-)

**Comment by g-plane** [comment]
So you can publish it to npm.

**Comment by platinumazure** [comment]
I don't think we're likely to accept new formatters at this point unless there's a clear need for it in order for ESLint to work with popular integrations. Others on the team might disagree, though.

As noted, you can publish the formatter to npm and then others can still get the benefit. :smile:
